tools/azure-devtools/src/azure_devtools/scenario_tests/preparers.py

```diff
@@ -36,20 +50,59 @@ def _prepare_create_resource(self, test_class_instance, **kwargs):
         else:
             resource_name = self.moniker
 
-        with self.override_disable_recording():
-            retries = 4
-            for i in range(retries):
-                try:
-                    parameter_update = self.create_resource(
-                        resource_name,
-                        **kwargs
-                    )
-                    break
-                except AzureNameError:
-                    if i == retries - 1:
+        # If a child is cached we must use the same cached resource their equivalent parent did so all the deps line up
+        try:
+            child_is_cached = getattr(fn, '__use_cache')
+            # Note: If it is ever desired to make caching inferred, remove this if/throw.
+            # This ensures that a user must _very specifically say they want caching_ on an item and all parents.
+            if not self._use_cache:
```

Comment:
Note to self:  add "child_is_cached" to this if; even though it's implied in getting to this point since getattr would fail, being explicit would improve code clarity and robustness.